api-generation/src/main/scala/ai/h2o/sparkling/api/generation/scala/MOJOModelTemplate.scala

```diff
@@ -21,23 +21,31 @@ import ai.h2o.sparkling.api.generation.common._
 import ai.h2o.sparkling.api.generation.scala.ParametersTemplate.resolveParameterConstructorMethodType
 
 object MOJOModelTemplate
-  extends ((AlgorithmSubstitutionContext, ParameterSubstitutionContext) => String)
+  extends ((AlgorithmSubstitutionContext, ParameterSubstitutionContext, ModelOutputSubstitutionContext) => String)
   with ParametersTemplateBase
   with ScalaEntityTemplate
-  with ParameterResolver {
+  with ParameterResolver
+  with OutputResolver {
 
   def apply(
       algorithmSubstitutionContext: AlgorithmSubstitutionContext,
-      parameterSubstitutionContext: ParameterSubstitutionContext): String = {
+      parameterSubstitutionContext: ParameterSubstitutionContext,
+      outputSubstitutionContext: ModelOutputSubstitutionContext): String = {
 
     val parameters = resolveParameters(parameterSubstitutionContext)
-      .filter(parameter =>
-        !IgnoredParameters.ignoredInMOJOs(algorithmSubstitutionContext.entityName).contains(parameter.h2oName))
+      .filterNot(parameter =>
+        IgnoredParameters.ignoredInMOJOs(algorithmSubstitutionContext.entityName).contains(parameter.h2oName))
+
+    val outputs = resolveOutputs(outputSubstitutionContext)
+      .filterNot(output => IgnoredOutputs.all(algorithmSubstitutionContext.entityName).contains(output.h2oName))
+      .filterNot(output => IgnoredOutputs.ignoredTypes(output.dataType.getSimpleName))
 
     val explicitFieldImplementations = parameterSubstitutionContext.explicitFields.flatMap(_.mojoImplementation) ++
       parameterSubstitutionContext.deprecatedFields.flatMap(_.mojoImplementation)
 
     val imports = Seq(
+      "scala.collection.JavaConverters._",
+      "com.google.gson.JsonObject",
```

Comment:
btw. not directly related to that PR but when I was doing some changes in that class I was thinking if there's a reason we use GSON here and not some Scala specific library as it's a bit hard to work with sometimes

Reply:
H2O-3 already uses GSON for json parsing. (see https://github.com/h2oai/h2o-3/blob/879e58d3e5f98064b738189febecd845bf4f87c4/h2o-genmodel/src/main/java/hex/genmodel/ModelMojoReader.java). I think the main motivation why we didn't go for another lib was that GSON is already on driver and executor classpath.

Reply:
that's what I thought :) maybe (just maybe) we could consider json4s in the future if we ever have time for improvements as I see spark uses that